src/System.Private.Xml/src/System/Xml/BinHexDecoder.cs

```diff
@@ -164,7 +175,11 @@ public static unsafe byte[] Decode(char[] chars, bool allowOddChars)
 
             if (hasHalfByteCached && !allowOddChars)
             {
+#if XMLSERIALIZERGENERATOR
+                throw new XmlException(SR.Format(SR.Xml_InvalidBinHexValueOddCount, new string(chars)));
+#else
                 throw new XmlException(SR.Xml_InvalidBinHexValueOddCount, new string(chars));
+#endif
```

Comment:
We can override the path to the .resx file by setting StringResourcesPath in csproj. See https://github.com/dotnet/corefx/blob/master/Tools-Override/Build.Common.targets#L177 for details.